src/components/App/App.js

```diff
@@ -29,10 +29,21 @@ class App extends React.Component {
     const notes = [...this.state.notes, note];
     this.setState({ notes });
   }
-  deleteNote(key) {
-    const notes = [...this.state.notes];
-    notes[key] = null;
+  handleListDeleteNote(key) {
+    let notes = [...this.state.notes];
+    notes.splice(key, 1);
+    if (notes.length === 0) {
+      notes = [getDefaultNote()];
+    }
     this.setState({ notes });
+    this.refreshFocus(key);
```

Comment:
這裡建議放在 setState 第二個參數 callback 裡，setState 不一定會立即和 state 同步

https://facebook.github.io/react/docs/react-component.html#setstate
> setState() does not always immediately update the component. It may batch or defer the update until later. This makes reading this.state right after calling setState() a potential pitfall. Instead, use componentDidUpdate or a setState callback (setState(updater, callback)), either of which are guaranteed to fire after the update has been applied.

Reply:
感謝您的寶貴提醒，已修正於 #1517bc4。